### ISSUE TITLE
Bazel warns the genrule `//example:base` which outputs a directory 

### ISSUE THREAD

**Issue by tetsuok** [bazel-contrib/rules_oci#65]
When I build `//example:base` and other targets that depend on it, Bazel warns the output of `//example:base` is a directory as follows:

```shell
$ bazel build //example:base
...
WARNING: /Volumes/work/github/rules_oci/example/BUILD.bazel:25:8: output 'example/layout' of //example:base is a directory; dependency checking of directories is unsound
```

Build itself succeeds, but it would be nice to fix the warning or work around the underlying issue.

I see the similar warning when I ran `bazel build //src:image` in `e2e/js_image_oci` of rules_js (Probably, the root cause of the warning is same):

```text
WARNING: /private/var/tmp/_bazel_t/ce0e746d0a64bfce50fc270eb1254184/external/debian_amd64/BUILD.bazel:2:8: output 'external/debian_amd64/layout' of @debian_amd64//:image is a directory; dependency checking of directories is unsound
```

**Comment by alexeagle** [comment]
I don't think we can do anything about this. We use `genrule` in that example. https://docs.aspect.build/rules/aspect_bazel_lib/docs/run_binary/ knows how to produce directory outputs, but doesn't permit a bash one-liner like we use in that example.

I think the upside is that users would generally never use a genrule. Maybe we could even remove that from our examples, or make an `examples/genrule` folder that is documented as "produces a warning from Bazel, don't do it this way unless you have reason X or Y"

**Comment by alexeagle** [comment]
@thesayyn I think we should just change //examples:base to use our normal oci_pull mechanism now, this was using `crane pull` in a genrule for historical reasons.

**Comment by thesayyn** [comment]
> @thesayyn I think we should just change //examples:base to use our normal oci_pull mechanism now, this was using `crane pull` in a genrule for historical reasons.

done in #165 but i needed to fix #151 so working on it.
